crates/sui-bridge-indexer/src/storage.rs

```diff
@@ -196,36 +215,40 @@ impl IndexerProgressStore for PgBridgePersistent {
         Ok(None)
     }
 
-    async fn get_ongoing_tasks(&self, prefix: &str) -> Result<Tasks, anyhow::Error> {
+    async fn get_ongoing_tasks(&self, prefix: &str) -> Result<Tasks, Error> {
+        use schema::progress_store::dsl::*;
         let mut conn = self.pool.get().await?;
         // get all unfinished tasks
-        let cp: Vec<models::ProgressStore> =
+        let cp = progress_store
             // TODO: using like could be error prone, change the progress store schema to stare the task name properly.
-            QueryDsl::filter(
-                QueryDsl::filter(dsl::progress_store, columns::task_name.like(format!("{prefix} - %"))),
-                columns::checkpoint.lt(columns::target_checkpoint))
-            .order_by(columns::target_checkpoint.desc())
-            .load(&mut conn)
+            .filter(task_name.like(format!("{prefix} - %")))
+            .filter(checkpoint.lt(target_checkpoint))
+            .order_by(target_checkpoint.desc())
+            .load::<ProgressStore>(&mut conn)
             .await?;
         let tasks = cp.into_iter().map(|d| d.into()).collect();
         Ok(Tasks::new(tasks)?)
     }
 
-    async fn get_largest_backfill_task_target_checkpoint(
-        &self,
-        prefix: &str,
-    ) -> Result<Option<u64>, Error> {
+    async fn get_largest_indexed_checkpoint(&self, prefix: &str) -> Result<Option<u64>, Error> {
+        use schema::progress_store::dsl::*;
         let mut conn = self.pool.get().await?;
-        let cp: Option<i64> =
+        let cp = progress_store
             // TODO: using like could be error prone, change the progress store schema to stare the task name properly.
-            QueryDsl::filter(QueryDsl::filter(dsl::progress_store
-                .select(columns::target_checkpoint), columns::task_name.like(format!("{prefix} - %"))),
-                columns::target_checkpoint.ne(i64::MAX))
-            .order_by(columns::target_checkpoint.desc())
+            .filter(task_name.like(format!("{prefix} - %")))
+            .filter(target_checkpoint.eq(i64::MAX))
+            .select(checkpoint)
```

Comment:
do we need order by desc here?

Reply:
We only have one live task in the progress store (this query filter by target cp equals i64::Max, which is the live task), so will always only have 1 entry.